### PR TITLE
[TASK] Cleanup old code

### PR DESCRIPTION
**Pull request by xf-**
- Only support array for dataPath. We convert it in anyway and test it again.

## Checklist:

- [x] My code follows the code style of this project. (CI will test it anyway and also needs approval)

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -70,21 +66,13 @@ define(['moment', 'utils/router', 'leaflet', 'gui', 'helper', 'utils/language'],
...
-      if (typeof config.dataPath === 'string' || config.dataPath instanceof String) {
-        config.dataPath = [config.dataPath];
-      }
-
-      for (var i in config.dataPath) {
-        if (config.dataPath.hasOwnProperty(i)) {
-          urls.push(config.dataPath[i] + 'meshviewer.json');
-        }
-      }
+      config.dataPath.forEach(function (d, i) {
```
Comment (genofire) [inline lib/main.js:78]: need change of `config.json` now it has always to be an array of urls

File: lib/utils/router.js [hunk 2]
```diff
@@ -16,17 +16,20 @@ define(['Navigo'], function (Navigo) {
...
         targets.forEach(function (t) {
-          t.gotoNode(objects.nodes[d.nodeId], objects.nodeDict);
+          t.gotoNode(objects.nodeDict[d.nodeId], objects.nodeDict);
         });
       }
     }
 
     function gotoLink(d) {
-      if (d.linkId in objects.links) {
+      var link = objects.links.find(function (value) {
```
Comment (genofire) [inline lib/utils/router.js:27]: Idea: performance question: what happen more often?
 * foreach in setData every minute
 * gotoLink find the link

**Review by genofire — CHANGES_REQUESTED** [review]
`config.json` need to fixed

File: lib/main.js (hunk 1 above)
Comment (xf-) [reply lib/main.js:78]: yeah, nice catch, didn't submit the testurl.

File: lib/utils/router.js (hunk 2 above)
Comment (xf-) [reply lib/utils/router.js:27]: goToLink is used then url has a link. Max 1 per minute update or by clicking a link. Lookup takes 0.1 millisecond with 1000 links.

File: config.json [hunk 3]
```diff
@@ -23,8 +23,10 @@
       "title": "Jahresstatistik - weiteren Statistiken"
     }
   ],
-  // String or array of data provider are supported
-  "dataPath": "https://regensburg.freifunk.net/data/",
+  // Array of data provider are supported
+  "dataPath": [
+    "https://regensburg.freifunk.net/data/"
```
Comment (genofire) [inline config.json:28]: could we use `./` or `./data` in future and develop branch?
you still have ffrgb-config branch

**Review by genofire — APPROVED** [review]

File: config.json (hunk 3 above)
Comment (xf-) [reply config.json:28]: no - the idea behind this is to load git and build and it works.
We have a origin policy (json works) and also fixedCenter, grafana is on Regensburg.

Comment (genofire) [reply config.json:28]: okay

File: lib/forcegraph.js [hunk 4]
```diff
@@ -198,7 +198,7 @@ define(['d3-selection', 'd3-force', 'd3-zoom', 'd3-drag', 'd3-timer', 'd3-ease',
       });
 
       self.setData = function setData(data) {
-        intNodes = data.nodes.all.map(function (d) {
```
Comment (genofire) [inline lib/forcegraph.js:201]: bin mir nicht sicher, ob das gut geht -> dieser umweg wurde ja genommen, um die position im forcegraph zu speichern.

Ansonsten fehlt im forcegraph gotoNode ...

File: lib/nodelist.js [hunk 5]
```diff
@@ -85,15 +85,14 @@ define(['sorttable', 'snabbdom', 'helper'], function (SortTable, V, helper) {
...
-        if (e.is_online) {
-          n.uptime = d.now - new Date(e.uptime).getTime();
-        } else {
-          n.uptime = e.lastseen - d.now;
-        }
-        n.neighbours = e.neighbours;
-        return n;
+      var data = [];
+      d.nodes.online.map(function (e) {
+        data.push(Object.create(e));
```
Comment (genofire) [inline lib/nodelist.js]: mach doch bitte vorher eine variable, überschreib uptime und dann push.
(sieht complex mit length aus, obwohl es das ja nicht ist)

File: lib/nodelist.js [hunk 6]
```diff
@@ -92,7 +92,6 @@ define(['sorttable', 'snabbdom', 'helper'], function (SortTable, V, helper) {
         } else {
           n.uptime = e.lastseen - d.now;
         }
-        n.neighbours = e.neighbours;
```
Comment (genofire) [inline lib/nodelist.js:95]: warum? wird doch in zeile 72 benötigt

Comment (genofire) [reply lib/nodelist.js:95]: ups, wird ja gecopies

File: lib/forcegraph.js (hunk 4 above)
Comment (genofire) [reply lib/forcegraph.js:201]: https://github.com/ffrgb/meshviewer/commit/470785710c75817e7d5824469782b11b90f6ddfb#diff-5c9339ed3752f7f0d8ac2273b5114de7R247

Comment (genofire) [reply lib/forcegraph.js:201]: und beim highlighten forcegraph/draw.js

**Pull request merged**